python/tank/authentication/saml2_sso.py

```diff
@@ -0,0 +1,605 @@
+                    # This is likely because the subdomain is not valid.
+                    # e.g. https://foobar.shotgunstudio.com
+                    # Here the domain (shotgunstudio.com) is valid, but not
+                    # foobar.
+                    session.error = HTTP_CANT_CONNECT_TO_SHOTGUN
+                else:
+                    pass
+            elif error is QtNetwork.QNetworkReply.NetworkError.UnknownContentError:
+                # This means that the site does not support SSO or that
+                # it is not enabled.
+                session.error = HTTP_AUTHENTICATE_SSO_NOT_UPPORTED
+            elif error is QtNetwork.QNetworkReply.NetworkError.ContentOperationNotPermittedError:
+                # This means that the SSO login worked, but that the user does
+                # have access to the site.
+                session.error = HTTP_CANT_AUTHENTICATE_SSO_NO_ACCESS
+            else:
+                session.error = reply.attribute(QtNetwork.QNetworkRequest.HttpReasonPhraseAttribute)
+                log.error('==- on_http_response_finished: %s - %s' % (reply.url(), reply.error()))
+        elif url.startswith(session.host + URL_LOGIN_PATH):
+            # If we are being redirected to the login page, then SSO is not
+            # enabled on that site.
+            session.error = HTTP_AUTHENTICATE_SSO_NOT_UPPORTED
+
+        if session.error:
+            # If there are any errors, we exit by force-closing the dialog.
+            self._dialog.accept()
+
+    def is_handling_event(self):
+        """
+        Called to know if an event is currently being handled.
+        """
+        # log.debug('==- is_handling_event')
+        return self._event_data is not None
+
+    def handle_event(self, event_data):
+        """
+        Called to start the handling of an event.
+        """
+        log.debug('==- handle_event')
+
+        if not self.is_handling_event():
+            self._event_data = event_data
+
+            # At this point, we want to stop any background session renewal, as
+            # it may impede with our new sesion login.
+            self.stop_session_renewal()
+
+            self.start_new_session(event_data)
+        else:
+            log.error('Calling handle_event while event %s is currently being handled' % self._event_data['event'])
+
+    def resolve_event(self, end_session=False):
+        """
+        Called to return the results of the event.
+        """
+        log.debug('==- resolve_event')
+
+        if self.is_handling_event():
+            # rvc.sendInternalEvent(self._event_data['event'], self._session.assembleSession())
+            if end_session:
+                self.end_current_session()
+            self._event_data = None
+        else:
+            log.error('Called resolve_event when no event is being handled.')
+
+    ############################################################################
```

Comment:
please remove - we don't typically do this in tk-core. nice if this file could look like the existing python files